boot/boot.go

```diff
@@ -429,8 +429,50 @@ func updateManagedBootConfigForBootloader(dev Device, mode, gadgetSnapOrDir stri
 		return false, err
 	}
 	// boot config update can lead to a change of kernel command line
-	if err := observeCommandLineUpdate(dev.Model(), gadgetSnapOrDir); err != nil {
+	_, err = observeCommandLineUpdate(dev.Model(), commandLineUpdateReasonSnapd, gadgetSnapOrDir)
+	if err != nil {
 		return false, err
 	}
 	return tbl.UpdateBootConfig()
 }
+
+// UpdateCommandLineForGadgetComponent handles the update of a gadget that
+// contributes to the kernel command line of the run system. Returns true when a
+// change in command line has been observed and a reboot is needed. The reboot,
+// if needed, should be requested as the the earliest possible occasion.
+func UpdateCommandLineForGadgetComponent(dev Device, gadgetSnapOrDir string) (needsReboot bool, err error) {
+	if !dev.HasModeenv() {
+		// only UC20 devices are supported
+		return false, fmt.Errorf("internal error: command line component cannot be updated on non UC20 devices")
+	}
+	opts := &bootloader.Options{
+		Role:        bootloader.RoleRunMode,
+		NoSlashBoot: true,
```

Comment:
do we need to use this?  won't this always be called from run mode? as it's about updates?